MapboxNavigation/LanesView.swift

```diff
@@ -67,30 +67,28 @@ open class LanesView: UIView {
     }
     
     func update(for currentLegProgress: RouteLegProgress) {
-        guard let step = currentLegProgress.upComingStep else { return }
-        guard !currentLegProgress.userHasArrivedAtWaypoint else { return }
-        let durationRemaining = currentLegProgress.currentStepProgress.durationRemaining
-        
         clearLaneViews()
         
-        if let allLanes = step.intersections?.first?.approachLanes,
-            let usableLanes = step.intersections?.first?.usableApproachLanes,
-            durationRemaining < RouteControllerMediumAlertInterval {
-            
-            for (i, lane) in allLanes.enumerated() {
-                let laneView = laneArrowView()
-                laneView.lane = lane
-                laneView.maneuverDirection = step.maneuverDirection
-                laneView.isValid = usableLanes.contains(i as Int)
-                stackView.addArrangedSubview(laneView)
-            }
+        let step = currentLegProgress.currentStep
+        let spokenInstructionIndex = currentLegProgress.currentStepProgress.spokenInstructionIndex
+        guard let visualInstructions = step.instructionsDisplayedAlongStep, spokenInstructionIndex < visualInstructions.count else {
+            hide()
+            return
         }
         
-        if stackView.arrangedSubviews.count > 0 {
-            show()
-        } else {
-            hide()
+        let currentInstruction = visualInstructions[spokenInstructionIndex]
```

Comment:
It looks like we're still using `spokenInstructionIndex ` instead of the visual index.